trillian/integration/ct_integration.go

```diff
@@ -980,3 +981,33 @@ func setTreeState(ctx context.Context, adminServer string, logID int64, state tr
 	}
 	return nil
 }
+
+// NotAfterForLog returns a NotAfter time to be used for certs submitted
+// to the given log instance, allowing for any temporal shard configuration.
+func NotAfterForLog(c *configpb.LogConfig) (time.Time, error) {
+	if c.NotAfterStart == nil && c.NotAfterLimit == nil {
+		return time.Now().Add(24 * time.Hour), nil
+	}
+
+	if c.NotAfterStart != nil {
+		start, err := ptypes.Timestamp(c.NotAfterStart)
+		if err != nil {
+			return time.Time{}, fmt.Errorf("failed to parse NotAfterStart: %v", err)
+		}
+		if c.NotAfterLimit == nil {
+			return start.Add(24 * time.Hour), nil
+		}
+
+		limit, err := ptypes.Timestamp(c.NotAfterLimit)
+		if err != nil {
+			return time.Time{}, fmt.Errorf("failed to parse NotAfterLimit: %v", err)
+		}
+		return time.Unix(0, (limit.UnixNano()-start.UnixNano())/2+start.UnixNano()), nil
```

Comment:
At first this looked unsafe but probably isn't as seconds is zero. However it looks odd as it's treating times as durations? Could it be done without all the conversions though?

Also does it need to check start < limit here?

Reply:
Added a commit with a `time.Duration`-based calculation (plus some tests).

A log config with start>limit will get bounced by `ValidateLogConfig` so I think it's OK to skip a check here.

Reply:
Thanks, that is easier to read.